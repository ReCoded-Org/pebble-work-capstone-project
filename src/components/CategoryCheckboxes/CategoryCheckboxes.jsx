import InputComponent from "../InputComponent";
const CategoryCheckboxes = ({
    style = "mb-2 h-6 w-6 rounded text-secondary-300",
}) => {
    return (
        <div>
            <div className='relative w-full'>
                <div className='fixed left-0 right-0 bottom-0 z-10 h-4/5 overflow-y-scroll rounded-t-xl border border-black bg-white text-2xl shadow-lg scrollbar-thin scrollbar-thumb-secondary-300 sm:relative sm:h-56 sm:w-full sm:rounded sm:border-none sm:text-base'>
                    <div className='m-4 flex flex-col'>
                        <h1 className='text-center font-bold sm:hidden'>
                            Choose Category
                        </h1>
                        <form>
                            <InputComponent
                                type='checkbox'
                                name='all'
                                id='all'
                                Value='all'
                                label=' All'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='no-poverty'
                                id='no-poverty'
                                value='no-poverty'
                                label=' No Poverty'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='zero-hunger'
                                id='zero-hunger'
                                value='zero-hunger'
                                label=' Zero Hunger'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='good-health-and-well-being'
                                id='good-health-and-well-being'
                                value='good-health-and-well-being'
                                label=' Good Health and Well-Being'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='quality-education'
                                id='quality-education'
                                value='quality-education'
                                label=' Quality Education'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='gender-equality'
                                id='gender-equality'
                                value='gender-equality'
                                label=' Gender Equality'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='clean-water-and-sanitation'
                                id='clean-water-and-sanitation'
                                value='clean-water-and-sanitation'
                                label=' Clean Water and Sanitation'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='affordable-and-clean-energy'
                                id='affordable-and-clean-energy'
                                value='affordable-and-clean-energy'
                                label=' Affordable and Clean Energy'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='decent-work-and-economic-growth'
                                id='decent-work-and-economic-growth'
                                value='decent-work-and-economic-growth'
                                label=' Decent Work and Economic Growth'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='industry-innovation-and-infrastructure'
                                id='industry-innovation-and-infrastructure'
                                value='industry-innovation-and-infrastructure'
                                label=' Industry, Innovation and Infrastructure'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='reduced-inequalities'
                                id='reduced-inequalities'
                                value='reduced-inequalities'
                                label=' Reduced Inequalities'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='sustainable-cities-and-communities'
                                id='sustainable-cities-and-communities'
                                value='sustainable-cities-and-communities'
                                label=' Sustainable Cities and Communities'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='responsible-consumption-and-production'
                                id='responsible-consumption-and-production'
                                value='responsible-consumption-and-production'
                                label=' Responsible Consumption and Production'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='climate-action'
                                id='climate-action'
                                value='climate-action'
                                label=' Climate Action'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='life-below-water'
                                id='life-below-water'
                                value='life-below-water'
                                label=' Life Below Water'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='life-on-land'
                                id='life-on-land'
                                value='life-on-land'
                                label=' Life on Land'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='peace-and-justice-and-strong-institutions'
                                id='peace-and-justice-and-strong-institutions'
                                value='peace-and-justice-and-strong-institutions'
                                label=' Peace and Justice and Strong Institutions'
                                className={style}
                            />
                            <br />
                            <InputComponent
                                type='checkbox'
                                name='partnership-for-the-goals'
                                id='partnership-for-the-goals'
                                value='partnership-for-the-goals'
                                label=' Partnership for the Goals'
                                className={style}
                            />
                            <br />
                        </form>
                        <div className='fixed left-1/4 bottom-3 sm:static sm:bottom-0 sm:mx-auto'>
                            <form>
                                <InputComponent
                                    type='Submit'
                                    className='mt-2 w-48 rounded-md border-2 bg-secondary-300 text-white'
                                    value='Apply Filters'
                                />
                            </form>
                        </div>
                        <div className='my-5 sm:hidden'></div>
                    </div>
                </div>
            </div>
        </div>
    );
};

export default CategoryCheckboxes;
