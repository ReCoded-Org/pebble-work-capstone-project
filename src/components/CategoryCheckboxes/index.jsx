import CategoryCheckboxes from "./CategoryCheckboxes";
export default CategoryCheckboxes;
