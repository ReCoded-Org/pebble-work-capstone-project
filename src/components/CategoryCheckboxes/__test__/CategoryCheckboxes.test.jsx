import renderer from "react-test-renderer";

import CategoryCheckboxes from "../CategoryCheckboxes";

it("renders correctly", () => {
    const tree = renderer.create(<CategoryCheckboxes />).toJSON();
    expect(tree).toMatchSnapshot();
});
