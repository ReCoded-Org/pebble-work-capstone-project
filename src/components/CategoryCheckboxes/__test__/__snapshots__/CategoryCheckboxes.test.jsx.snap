// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="relative w-full"
  >
    <div
      className="fixed left-0 right-0 bottom-0 z-10 h-4/5 overflow-y-scroll rounded-t-xl border border-black bg-white text-2xl shadow-lg scrollbar-thin scrollbar-thumb-secondary-300 sm:relative sm:h-56 sm:w-full sm:rounded sm:border-none sm:text-base"
    >
      <div
        className="m-4 flex flex-col"
      >
        <h1
          className="text-center font-bold sm:hidden"
        >
          Choose Category
        </h1>
        <form>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="all"
              name="all"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="all"
            />
            <label
              htmlFor="all"
            >
               All
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="no-poverty"
              name="no-poverty"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="no-poverty"
            />
            <label
              htmlFor="no-poverty"
            >
               No Poverty
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="zero-hunger"
              name="zero-hunger"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="zero-hunger"
            />
            <label
              htmlFor="zero-hunger"
            >
               Zero Hunger
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="good-health-and-well-being"
              name="good-health-and-well-being"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="good-health-and-well-being"
            />
            <label
              htmlFor="good-health-and-well-being"
            >
               Good Health and Well-being
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="quality-education"
              name="quality-education"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="quality-education"
            />
            <label
              htmlFor="quality-education"
            >
               Quality Education
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="gender-equality"
              name="gender-equality"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="gender-equality"
            />
            <label
              htmlFor="gender-equality"
            >
               Gender Equality
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="clean-water-and-sanitation"
              name="clean-water-and-sanitation"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="clean-water-and-sanitation"
            />
            <label
              htmlFor="clean-water-and-sanitation"
            >
               Clean Water and Sanitation
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="affordable-and-clean-energy"
              name="affordable-and-clean-energy"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="affordable-and-clean-energy"
            />
            <label
              htmlFor="affordable-and-clean-energy"
            >
               Affordable and Clean Energy
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="decent-work-and-economic-growth"
              name="decent-work-and-economic-growth"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="decent-work-and-economic-growth"
            />
            <label
              htmlFor="decent-work-and-economic-growth"
            >
               Decent Work and Economic Growth
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="industry-innovation-and-infrastructure"
              name="industry-innovation-and-infrastructure"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="industry-innovation-and-infrastructure"
            />
            <label
              htmlFor="industry-innovation-and-infrastructure"
            >
               Industry, Innovation and Infrastructure
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="reduced-inequalities"
              name="reduced-inequalities"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="reduced-inequalities"
            />
            <label
              htmlFor="reduced-inequalities"
            >
               Reduced Inequalities
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="responsible-consumption-and-production"
              name="responsible-consumption-and-production"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="responsible-consumption-and-production"
            />
            <label
              htmlFor="responsible-consumption-and-production"
            >
               Responsible Consumption and Production
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="sustainable-cities-and-communities"
              name="sustainable-cities-and-communities"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="sustainable-cities-and-communities"
            />
            <label
              htmlFor="sustainable-cities-and-communities"
            >
               Sustainable Cities and Communities
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="climate-action"
              name="climate-action"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="climate-action"
            />
            <label
              htmlFor="climate-action"
            >
               Climate Action
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="life-below-water"
              name="life-below-water"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="life-below-water"
            />
            <label
              htmlFor="life-below-water"
            >
               Life Below Water
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="life-on-land"
              name="life-on-land"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="life-on-land"
            />
            <label
              htmlFor="life-on-land"
            >
               Life on Land
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="peace-and-justice-and-strong-institutions"
              name="peace-and-justice-and-strong-institutions"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="peace-and-justice-and-strong-institutions"
            />
            <label
              htmlFor="peace-and-justice-and-strong-institutions"
            >
               Peace and Justice and Strong Institutions
            </label>
            <br />
          </div>
          <div>
            <input
              className="mb-2 h-6 w-6 rounded text-secondary-300"
              id="partnership-for-the-goals"
              name="partnership-for-the-goals"
              onChange={[Function]}
              placeholder="Your input here"
              type="checkbox"
              value="partnership-for-the-goals"
            />
            <label
              htmlFor="partnership-for-the-goals"
            >
               Partnership for the Goals
            </label>
            <br />
          </div>
        </form>
        <div
          className="fixed left-1/4 bottom-3 sm:static sm:bottom-0 sm:mx-auto"
        >
          <form
            onClick={[Function]}
          >
            <input
              className="mt-2 w-48 rounded-md border-2 bg-secondary-300 text-white"
              placeholder="Your input here"
              type="button"
              value="Apply Filters"
            />
            <label />
          </form>
        </div>
        <div
          className="my-5 sm:hidden"
        />
      </div>
    </div>
  </div>
</div>
`;
